--- etc/rc.d/rc.sysinit.orig	2011-08-16 11:11:57.000000000 -0700
+++ etc/rc.d/rc.sysinit	2011-08-16 11:12:07.000000000 -0700
@@ -581,7 +581,7 @@
 
 # Clean up /var.
 rm -rf /var/lock/cvs/* /var/run/screen/*
-find /var/lock /var/run ! -type d -exec rm -f {} \;
+find /var/lock/ /var/run/ ! -type d -exec rm -f {} \;
 rm -f /var/lib/rpm/__db* &> /dev/null
 rm -f /var/gdm/.gdmfifo &> /dev/null
 
--- etc/init/rcS.conf.orig	2012-01-14 13:27:21.000000000 +1100
+++ etc/init/rcS.conf	2012-01-14 13:29:25.000000000 +1100
@@ -13,6 +13,8 @@ task
 # without information so that it defaults to previous=N runlevel=S.
 console output
 pre-start script
+	/sbin/rc.preinit --run --exec "/bin/echo oneSIS: preinit complete"
+
 	for t in $(cat /proc/cmdline); do
 		case $t in
 			emergency)
